collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -360,6 +363,7 @@ private void updateIndex(boolean isAsync, @Nullable FailedValidationResult valid
     public void exit() {
         formSessionRepository.clear(sessionId);
         ReferenceManager.instance().reset();
+        changeLocks.getFormsLock().unlock();
```

Comment:
It doesn't seem like there's a test for the `unlock` call here.

Reply:
Done.